colorbleed/plugins/maya/publish/increment_current_file_deadline.py

```diff
@@ -0,0 +1,75 @@
+import pyblish.api
+
+
+class IncrementCurrentFileDeadline(pyblish.api.ContextPlugin):
+    """Submit available render layers to Deadline
+
+    Renders are submitted to a Deadline Web Service as
+    supplied via the environment variable AVALON_DEADLINE
+
+    """
+
+    label = "Increment current file"
+    order = pyblish.api.IntegratorOrder + 9.0
+    hosts = ["maya"]
+    families = ["colorbleed.renderlayer"]
+
+    def process(self, context):
+
+        from maya import cmds
+
+        from colorbleed.action import get_errored_plugins_from_data
+
+        plugins = get_errored_plugins_from_data(context)
+
+        if any(plugin.__name__ == "MindbenderSubmitDeadline"
+                for plugin in plugins):
+            raise RuntimeError("Skipping incrementing current file because"
+                               "submission to deadline failed.")
+
+        new_filename = self.version_up(context.data["currentFile"])
+
+        cmds.file(rename=new_filename)
+        cmds.file(save=True, force=True, type="mayaAscii")
+
+    def version_up(self, filepath):
```

Comment:
Note for selves: I'm expecting this kind of versioning logic to be available through the library or api so we can use it consistently in multiple locations. :)